kedro/io/data_catalog.py

```diff
@@ -30,6 +30,8 @@
 CREDENTIALS_KEY = "credentials"
 WORDS_REGEX_PATTERN = re.compile(r"\W+")
 
+Patterns = dict[str, dict[str, Any]]
```

Comment:
Is this an overkill? I found that we don't do this in `kedro` other than datasets which use `DI_` and `DO_` for data input/output. In this case I do think it has value as it helps a lot to read the code especially when we have a lot of dict of dict flowing around everywhere, I think this put some context.

@astrojuanlu WDYT about using type alias?

Reply:
I think this is a good use of a type alias! Not sure the lowercase `dict` will work for them (but if they do, fantastic - I sometimes get lost with Python typing)